src/main/java/fi/aalto/cs/apluscourses/intellij/model/ScalaSdk.java

```diff
@@ -0,0 +1,104 @@
+package fi.aalto.cs.apluscourses.intellij.model;
+
+import com.intellij.openapi.roots.libraries.LibraryProperties;
+import com.intellij.openapi.roots.libraries.PersistentLibraryKind;
+import com.intellij.openapi.util.io.FileUtilRt;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.Arrays;
+import net.lingala.zip4j.ZipFile;
+import org.apache.commons.io.FileUtils;
+import org.jetbrains.annotations.CalledWithWriteLock;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.plugins.scala.project.ScalaLanguageLevel;
+import org.jetbrains.plugins.scala.project.ScalaLibraryProperties;
+import org.jetbrains.plugins.scala.project.ScalaLibraryPropertiesState;
+import org.jetbrains.plugins.scala.project.ScalaLibraryType;
+
+public class ScalaSdk extends IntelliJLibrary<PersistentLibraryKind<ScalaLibraryProperties>,
+    ScalaLibraryPropertiesState> {
+
+  private static final String URL = "https://scala-lang.org/files/archive/";
+  private static final String[] ALL_CLASSES = {
+      "scala-compiler.jar",
+      "scala-library.jar",
+      "scala-reflect.jar"
+  };
+  private static final String[] CLASSES = {
+      "scala-library.jar",
+      "scala-reflect.jar"
+  };
+
+  private final String scalaVersion;
+
+  /**
+   * Constructs a new Scala SDK object.
+   *
+   * @param name Name that must match scala-sdk-0.0.0 pattern.
+   * @param project The IntelliJ project.
+   */
+  public ScalaSdk(@NotNull String name, @NotNull APlusProject project, int state) {
+    super(name, project, state);
+
+    scalaVersion = name.replace("scala-sdk-", "");
+  }
+
+  @Override
+  public void fetch() throws IOException {
+    File tempZipFile = createTempFile();
+    fetchZipTo(tempZipFile);
+    extractZip(tempZipFile);
+  }
+
+  @NotNull
+  private File createTempFile() throws IOException {
+    return FileUtilRt.createTempFile(getFileName(), ".zip");
+  }
+
+  private void fetchZipTo(File file) throws IOException {
+    FileUtils.copyURLToFile(new URL(URL + getFileName() + ".zip"), file);
```

Comment:
this might be a good addition to some of the methods here: https://github.com/Aalto-LeTech/intellij-plugin/blob/7b9f14b1efe246bd1da75f921d3d1dca0eac07bd/src/main/java/fi/aalto/cs/apluscourses/utils/CoursesClient.java#L50